onepwd/plugins/action/upload_s3_secret.py

```diff
@@ -0,0 +1,156 @@
+                pass
+            else: 
+                print("Overwrite is set to False. No action taken.")
+                return {}
+        except:
+            print("Secret doesn't exist yet")
+            s3_secret_exists = False
+
+        # Template creation - BUCKET_NAME, ACCESS_KEY and ACCESS_SECRET
+        content = f'{{"k":"string","n":"bucket_name","t":"BUCKET_NAME","v":"{BUCKET_NAME}"}},{{"k":"concealed","n":"access_key","t":"ACCESS_KEY","v":"{ACCESS_KEY}"}},{{"k":"concealed","n":"access_secret","t":"ACCESS_SECRET","v":"{ACCESS_SECRET}"}}'
+        template = '{"sections":[{"fields":[' + str(content) +  ']}]}'
+        encoded_item = url64.encode(template)   
+        
+        # Upload Secret if no 's3' secret exists
+        if s3_secret_exists == False: 
+            print("Uploading secret as requested...")
+            command = onepwd.OnePwd.create_item(op, category, encoded_item, title, vault=vault, url=url)
+            print("Secret uploaded")
+
+            return {'changed': 'true',
+                    'exectued' : command}
+
+        # Overwrite wanted? Update Values   
+        if overwrite == True and s3_secret_exists == True: 
+            
+            # Test if values are alredy configured as requested
+            # svalue is a list with the secret fields [{'k':'string','n':'bucket_name','t':'BUCKET_NAME','v':'My-Bucket-name'}, ...]
+            svalue = onepwd.get_secret_values_list(op, item_name=title, vault=vault)
+            # True = nothing changed, False = Requested value is different from current value
+            check_bucket = True
+            check_secret = True
+            check_key = True
+            if BUCKET_NAME is not None: 
+                check_bucket = (svalue[0]['v'] == BUCKET_NAME)
+                if svalue[0]['v'] == BUCKET_NAME:
+                    print("BUCKET_NAME already set as requested")
+                else: 
+                    print("BUCKET_NAME is different")
+            if ACCESS_KEY is not None: 
+                check_key = (svalue[1]['v'] == ACCESS_KEY)
+                if svalue[1]['v'] == ACCESS_KEY:
+                    print("ACCESS_KEY already set as requested")
+                else: 
+                    print("ACCESS_KEY is different")
+            if ACCESS_SECRET is not None: 
+                check_secret = (svalue[2]['v'] == ACCESS_SECRET)
+                if svalue[2]['v'] == ACCESS_SECRET:
+                    print("ACCESS_SECRET already set as requested")
+                else: 
+                    print("ACCESS_SECRET is different")
+
+            # Update Secret if changes are present
+            if (check_bucket and check_key and check_secret) == False:
+                onepwd.OnePwd.update_item(op, title, vault=vault, BUCKET_NAME=BUCKET_NAME, ACCESS_KEY=ACCESS_KEY, ACCESS_SECRET=ACCESS_SECRET )
+                print("Secret updated...") 
+                return {'changed': 'true',
+                'executed' : 'update item command'}
```

Comment:
You should als return the updated item

Reply:
The update_item is different from the create_item. It can't be returned with json.loads(), which throws an error, and therefore  the "return run_op_command_in_shell(command)" does not return any value.